src/searchModal.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+import $ from 'jquery';
+import _ from 'lodash';
+import __ from 'i18n';
+import layoutTpl from 'ui/searchModal/tpl/layout';
+import 'ui/searchModal/css/searchModal.css';
+import component from 'ui/component';
+import 'ui/modal';
+import 'ui/datatable';
+
+/**
+ * Creates a search modal instance
+ * @param {object} config - search modal configuration
+ * @param {string} config.query - search query to be set and triggered on component creation
+ * @param {string} config.url - search endpoint
+ * @param {object} config.events - events hub
+ */
+export default function searchModalFactory(config) {
+    const instance = component().setTemplate(layoutTpl);
+    let searchInput = null;
+    let searchButton = null;
+    let clearButton = null;
+    let running = false;
+    instance.on('render', () => renderModal());
+    instance.on('destroy', () => destroyModal());
+
+    /**
+     * Creates search modal, inits template selectors, and triggers initial search
+     */
+    function renderModal() {
+        initModal();
+        initUiSelectors();
```

Comment:
Could it be moved to the `.on('init', ` event?

Reply:
I have checked, but on `init` event the component is not yet rendered, causing an error on `initModal()` when:

```
    function initModal() {
        instance
            .getElement()
            .addClass('modal')
            (...)
```
Because instance.getElement() returns `undefined`.